packages/phalanx/src/Phalanx_DataLayout_MDALayout_Def.hpp

```diff
@@ -51,603 +51,195 @@
-  m_dim_name.push_back(PHX::print<Tag2>());
-  m_dim_name.push_back(PHX::print<Tag3>());
-  m_dim_name.push_back(PHX::print<Tag4>());
-  m_dim_name.push_back(PHX::print<Tag5>());
-
-  m_size = size1 * size2 * size3 * size4 * size5 * size6;
-
-  m_identifier = this->createIdentifier();
-}
-
-//**********************************************************************
-template<typename Tag0, typename Tag1, typename Tag2, typename Tag3,
-	 typename Tag4, typename Tag5, typename Tag6, typename Tag7>
-PHX::MDALayout<Tag0,Tag1,Tag2,Tag3,Tag4,Tag5,Tag6,Tag7>::
-MDALayout(size_type size1, size_type size2, size_type size3, 
-	  size_type size4, size_type size5)
-{ 
-  PHX::Check_num_ctor_arguments_equal_to_num_template_arguments<Rank,5>();
-
-  m_dim_size[0] = size1;
-  m_dim_size[1] = size2;
-  m_dim_size[2] = size3;
-  m_dim_size[3] = size4;
-  m_dim_size[4] = size5;
-
-  m_dim_name.push_back(PHX::print<Tag0>());
-  m_dim_name.push_back(PHX::print<Tag1>());
-  m_dim_name.push_back(PHX::print<Tag2>());
-  m_dim_name.push_back(PHX::print<Tag3>());
-  m_dim_name.push_back(PHX::print<Tag4>());
-
-  m_size = size1 * size2 * size3 * size4 * size5;
-
-  m_identifier = this->createIdentifier();
-}
-
-//**********************************************************************
-template<typename Tag0, typename Tag1, typename Tag2, typename Tag3,
-	 typename Tag4, typename Tag5, typename Tag6, typename Tag7>
-PHX::MDALayout<Tag0,Tag1,Tag2,Tag3,Tag4,Tag5,Tag6,Tag7>::
-MDALayout(size_type size1, size_type size2, size_type size3, 
-	  size_type size4)
-{ 
-  PHX::Check_num_ctor_arguments_equal_to_num_template_arguments<Rank,4>();
-
-  m_dim_size[0] = size1;
-  m_dim_size[1] = size2;
-  m_dim_size[2] = size3;
-  m_dim_size[3] = size4;
-
-  m_dim_name.push_back(PHX::print<Tag0>());
-  m_dim_name.push_back(PHX::print<Tag1>());
-  m_dim_name.push_back(PHX::print<Tag2>());
-  m_dim_name.push_back(PHX::print<Tag3>());
-
-  m_size = size1 * size2 * size3 * size4;
+template<int N=0, typename CurrentExtent, typename... Extents>
+void setExtentsVariadic(std::size_t* e, CurrentExtent ce, Extents... extents)
+{
+  e[N] = ce;
 
-  m_identifier = this->createIdentifier();
+  if constexpr (sizeof...(extents) != 0) {
+    setExtentsVariadic<N+1>(e,extents...);
+  }
 }
 
 //**********************************************************************
-template<typename Tag0, typename Tag1, typename Tag2, typename Tag3,
-	 typename Tag4, typename Tag5, typename Tag6, typename Tag7>
-PHX::MDALayout<Tag0,Tag1,Tag2,Tag3,Tag4,Tag5,Tag6,Tag7>::
-MDALayout(size_type size1, size_type size2, size_type size3)
-{ 
-  PHX::Check_num_ctor_arguments_equal_to_num_template_arguments<Rank,3>();
-
-  m_dim_size[0] = size1;
-  m_dim_size[1] = size2;
-  m_dim_size[2] = size3;
-
-  m_dim_name.push_back(PHX::print<Tag0>());
-  m_dim_name.push_back(PHX::print<Tag1>());
-  m_dim_name.push_back(PHX::print<Tag2>());
-
-  m_size = size1 * size2 * size3;
+template<int N=0, typename CurrentTag, typename... Tags>
```

Comment:
Does the compiler allow a default value for the 1st arg without having defaults for the following one?

Reply:
Also, do you need the default?

Reply:
Good point! Seemed to compile fine but I removed the default anyway just to be safe. Was trying to make it easier for users, but this is really only meant for the specific internal implementation.